test/tag_table_bulider.go

```diff
@@ -0,0 +1,61 @@
+package test
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/git-town/git-town/test/helpers"
+)
+
+// TagTableBuilder collects data about tags in Git repositories
+// in the same way that our Gherkin tables describing tags in repos are organized.
+type TagTableBuilder struct {
+
+	// tagsToLocations stores data about what locations tags are.
+	//
+	// Structure:
+	//   tag1: [local]
+	//   tag2: [local, remote]
+	tagToLocations map[string]helpers.OrderedStringSet
+}
+
+// NewTagTableBuilder provides a fully initialized instance of TagTableBuilder.
+func NewTagTableBuilder() TagTableBuilder {
+	result := TagTableBuilder{
+		tagToLocations: make(map[string]helpers.OrderedStringSet),
+	}
+	return result
+}
+
+// Add registers the given tag from the given location into this table.
+func (builder *TagTableBuilder) Add(tag, location string) {
+	locations, exists := builder.tagToLocations[tag]
+	if exists {
+		builder.tagToLocations[tag] = locations.Add(location)
+	} else {
+		builder.tagToLocations[tag] = helpers.NewOrderedStringSet(location)
+	}
+}
+
+// AddMany registers the given tags from the given location into this table.
+func (builder *TagTableBuilder) AddMany(tags []string, location string) {
+	for _, tag := range tags {
+		builder.Add(tag, location)
+	}
+}
+
+// Table provides the data accumulated by this TagTableBuilder as a DataTable.
+func (builder *TagTableBuilder) Table() (result DataTable) {
+	result.AddRow("NAME", "LOCATION")
+	tags := make([]string, len(builder.tagToLocations))
```

Comment:
Nice preallocation at the correct length!

Reply:
Linter made me do it haha